commit 23e7a7944f3779155e2f6bbc831b544eb925f387
Author: Jason Wang <jasowang@redhat.com>
Date:   Fri Aug 27 17:15:06 2010 +0800

    KVM: pit: Do not check pending pit timer in vcpu thread
    
    Pit interrupt injection was done by workqueue, so no need to check
    pending pit timer in vcpu thread which could lead unnecessary
    unblocking of vcpu.
    
    Signed-off-by: Jason Wang <jasowang@redhat.com>
    Signed-off-by: Avi Kivity <avi@redhat.com>

diff --git a/arch/x86/kvm/i8254.c b/arch/x86/kvm/i8254.c
index ddeb2314b522..2ad40a4ddc34 100644
--- a/arch/x86/kvm/i8254.c
+++ b/arch/x86/kvm/i8254.c
@@ -232,15 +232,6 @@ static void pit_latch_status(struct kvm *kvm, int channel)
 	}
 }
 
-int pit_has_pending_timer(struct kvm_vcpu *vcpu)
-{
-	struct kvm_pit *pit = vcpu->kvm->arch.vpit;
-
-	if (pit && kvm_vcpu_is_bsp(vcpu) && pit->pit_state.irq_ack)
-		return atomic_read(&pit->pit_state.pit_timer.pending);
-	return 0;
-}
-
 static void kvm_pit_ack_irq(struct kvm_irq_ack_notifier *kian)
 {
 	struct kvm_kpit_state *ps = container_of(kian, struct kvm_kpit_state,